extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php

```diff
@@ -158,22 +160,26 @@ private function renderGroup( $groupData ) {
 	private function renderInfoboxHero( $data ) {
 		$helper = new PortableInfoboxRenderServiceHelper();
 
+		// In Mercury SPA content of the first infobox's hero module has been moved to the article header.
+		$firstInfoboxAlredyRendered = \Wikia\PortableInfobox\Helpers\PortableInfoboxDataBag::getInstance()
+			->isFirstInfoboxAlredyRendered();
+
 		if ( array_key_exists( 'image', $data ) ) {
 			$image = $data[ 'image' ][ 0 ];
 			$image[ 'context' ] = self::MEDIA_CONTEXT_INFOBOX_HERO_IMAGE;
 			$image = $helper->extendImageData( $image, PortableInfoboxRenderServiceHelper::MOBILE_THUMBNAIL_WIDTH );
 			$data[ 'image' ] = $image;
 
 			if ( !$helper->isMercury() ) {
-				$markup = $this->renderItem( 'hero-mobile-wikiamobile', $data );
-			} else {
-				$markup = $this->renderItem( 'hero-mobile', $data );
+				return $this->renderItem( 'hero-mobile-wikiamobile', $data );
+			} elseif ( $firstInfoboxAlredyRendered ) {
+				return $this->renderItem( 'hero-mobile', $data );
 			}
-		} else {
-			$markup = $this->renderItem( 'title', $data[ 'title' ] );
+		} elseif ( !$helper->isMercury() || $firstInfoboxAlredyRendered ) {
+			return $this->renderItem( 'title', $data[ 'title' ] );
 		}
 
-		return $markup;
+		return '';
```

Comment:
This variable could be useful if anything would be additionally done to the `$markup` just before returning it.